llm-openai.el

```diff
@@ -73,20 +74,22 @@ MODEL is the embedding model to use, or nil to use the default.."
 
 (defun llm-openai--embedding-extract-response (response)
   "Return the embedding from the server RESPONSE."
-  (cdr (assoc 'embedding (aref (cdr (assoc 'data response)) 0))))
+  (let ((body (plz-response-body response)))
```

Comment:
Here and everywhere else, why not just pass in the body as the response?  I'd prefer the processing code to be agnostic to the request-handling infrastructure if possible.

Reply:
The gateway/proxy I have to deal with is not very generous when it comes to error messages in the HTTP response body. For example, there can be proxy gateway errors, like an auth token missing, or certificate missing/invalid. Those error responses are different from the OpenAI errors. In order to handle them properly I need to look at the status code of an HTTP request as well, and not only at the body. Therefore I find it useful to have the full response available in a provider.

Reply:
Could those be dealt with in error handling code instead of body processing code?  That is, in `llm-request-plz` could we detect these when in the `:then` and call the error handling function?  It'd be nice if we only called the llm provider code that parses responses when those responses are valid and there's no errors, and for that it seems you really should just need to pass the body.

Reply:
I think users of llm-request-plz (or more specific, the HTTP functionality) should be able get at the status code and headers as easily as at the body. Why hide that away? Making a decision based on looking at status code or headers of a HTTP response is very useful. If `llm-request-plz-sync`, just returns the body, how can I make those decisions? Or say, there's a feature request to show the HTTP status code in `llm-openai--error-message`. That's a feature I would actually love to have :)

Reply:
But please, change things around as you need, if you don't agree.

Reply:
Thanks for explaining your thought process.  An example would really help me understand.  But even if I buy that having access to the response code is necessary, I don't think it should be passed in anything related to `plz`.  Instead, I'd prefer another arg to the callback, or perhaps a struct defined in `llm-request-plz`.

Let me change things for now, and I'm open to adding a request code if it really is needed.

Reply:
Yes, go for it

Reply:
Thanks - I've been a bit busy today and my be busy tomorrow, but I'll continue making changes soon.

Reply:
@ahyatt Thanks for the update. No worries. Would you like to take over the OpenAI provider as well? I'm not yet familiar with function calling which is the part that is missing. I would close this PR then?

Reply:
I could look into supporting the "other" streaming JSON format if you tell me which one it is. Can you provide curl dump of the response?

Reply:
Hi @ahyatt 

I'm looking into making a "media type" for the Ollama streaming format. It has it's own content-type called `application/x-ndjson`.

https://github.com/ollama/ollama/blob/main/docs/api.md#generate-request-streaming

Reply:
Yeah, let me take over Open AI.  For the other JSON format, I was just looking around again, and it looks like maybe `llm-gemini` and `llm-vertex` actually now use an event-stream format, but with just text and not json.  The format I described later (streaming JSON) is the previous format.  You can see details of the format at [their quickstart guide](https://ai.google.dev/tutorials/rest_quickstart).    It might be trivial to do that.  I'll attempt to tackle Open AI, Gemini, and Vertex in the next week.

Making a media type for Ollama seems like a good idea, thanks!